File: Perl/parse_tag_count.pl
#!/bin/perl

use Getopt::Long;

use strict;
use File::stat;
use Time::localtime;
use QCAnalysis;
use QCAnalysis::TagCount;
use QCAnalysis::DB;
use QCAnalysis::RunTable;

#use warning;

my %arguments = ();

my $opt_ret = GetOptions( \%arguments, 'input=s', 'db_config=s' );

print "Input: " . $arguments{input} . "\n";
#print "Sample ID: " . $arguments{sample} . "\n";
print "DB configuration: ". $arguments{db_config}."\n";
my $input = $arguments{input};

my $line;
my $module;
my $status;
my $config =  $arguments{db_config};

#my $analysis = QCAnalysis->new();
my $db = QCAnalysis::DB->new();

my @analysis =  QCAnalysis::RunTable->parse_file( $input); 


$db->connect($config);
foreach(@analysis){
   print "Anaysis: ".$_;
   my $fast_qc_file = $_->get_property("path_to_counts");
	
   QCAnalysis::TagCount->parse_file($fast_qc_file, $_);
   $db->insert_analysis($_);
}
$db->disconnect();

File: python/parse_tag_count.py
import QCAnalysis
import TagCount, DB, RunTable

import os.sys

sys.path.append('/usr/users/ga002/shrestha/Dropbox/')

parser=optparse.OptionParser()
parser.add_option("-i", "--input", action="store", type="string", dest="input")
parser.add_option("--db_config", action="store", type="string", dest="db_config")

(options, args) = parser.parse_args()   #gets the arguments and options
print "Input: " + options['input'] + "\n"
print "DB configuration: " + options['db_config'] + "\n"
input=options['input']
line=""; module=""; status=""; config=options['db_config']

db=QCAnalysis.DB()

analysis=QCAnalysis.RunTable.parse_file(input)
db.connect()
for each in analysis:
    print "Analysis: " + each
    fast_qc_file = each.get_property("path_to_counts")
    QCAnalysis.TagCount.parse_file(fast_qc_file, each)
    db.insert_analysis(each)

db.disconnect()
